grails-app/controllers/org/pih/warehouse/picklist/PicklistController.groovy

```diff
@@ -218,22 +219,42 @@ class PicklistController {
         }
     }
 
-    def importPickListItems(ImportDataCommand command) {
-        StockMovement stockMovement = stockMovementService.getStockMovement(params.id)
-        List<PickPageItem> pickPageItems = stockMovementService.getPickPageItems(params.id, null, null)
+    def importPickListItems(PicklistImportDataCommand command) {
 
-        MultipartFile importFile = command.importFile
-        if (importFile.isEmpty()) {
-            throw new IllegalArgumentException("File cannot be empty")
-        }
+        // Bind stock movement based on provided ID
+        command.stockMovement = stockMovementService.getStockMovement(params.id)
+
+        // Bind user-provided location or current location
+        command.location = command.location ?: Location.get(session.warehouse.id)
```

Comment:
(Just thinking out loud)

I wonder if in case of importing, we will ever need current logged in location :thinking: 
For example, case with supported activity **Allow Overpick** I belive we should check this activity not on current logged in location but on origin because the origin location should decide if it supports this activity.

This was just one example but I guess any future validation will involve either the **origin** or **destination** locations, the current logged in location will probably be a single case that will not allow importing items if `current location != origin`.

Do I have a point or am I going off the track?

Reply:
Yup, I agree with you. I also think we should base it on the origin in this case. I brought this up in the PicklistImportDataCommand too.